### PR TITLE
[14.0][ADD] contract_split module

### PR DESCRIPTION
**Pull request by damdam-s**
based on #961 for contract test fix
superseed #960

### CODE REVIEW

File: contract_split/tests/__init__.py
Comment (petrus-v) [inline contract_split/tests/__init__.py:1]: I suppose you need import to get your test launch by CI and get proper coverage !

File: contract_split/models/contract.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+class ContractLine(models.Model):
+    _inherit = "contract.line"
+
+    splitted_from_line_id = fields.Many2one(
+        comodel_name="contract.line",
+        readonly=True,
+    )
+    splitted_from_contract_id = fields.Many2one(
+        comodel_name="contract.contract",
```
Comment (petrus-v) [inline contract_split/models/contract.py:53]: We could probably use a related field here using `splitted_from_line_id.contract_id` ?

File: contract_split/models/contract.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    @api.model
+    def _get_values_create_split_contract(self, split_wizard):
+        return {
+            "name": self._get_contract_split_name(split_wizard),
+            "partner_id": split_wizard.partner_id.id,
+            "invoice_partner_id": split_wizard.invoice_partner_id.id,
+            "original_contract_ids": [split_wizard.main_contract_id.id],
+            "line_recurrence": True,
+        }
```
Comment (petrus-v) [inline contract_split/models/contract.py:30]: I don't really understand why thoses methods are defined on the contract.contract ?

Probably I miss something but I've the feeling that it get more interest on the wizard and would make code easier to read ?!

File: contract_split/wizard/wizard_split_contract.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                contract_obj._get_values_create_split_contract(self)
+            )
+            # TODO: play onchange on partner_id. use onchange_helper from OCA ?
+            for line in self.split_line_ids:
+                original_line = line.original_contract_line_id
+                if not line.quantity_to_split:
+                    continue
+                if (
+                    float_compare(
+                        line.quantity_to_split, line.original_qty, precision_digits=2
```
Comment (petrus-v) [inline contract_split/wizard/wizard_split_contract.py]: should we avoid to hard code the precision and use the one defined on the unit of the product linked to the line ?

**Review by petrus-v — CHANGES_REQUESTED** [review]
I ask change mainly to ensure test are launched by CI otherwise mainly suggested improvement that I won't lock on it

**Review by mourad-ehm — APPROVED** [review]

File: contract_split/models/contract.py (hunk 1 above)
Comment (petrus-v) [reply contract_split/models/contract.py:53]: ok I realize if tomorrow we implement a move we know that split came from such contract !

**Review by petrus-v — APPROVED** [review]
Code reviews, looks good to me ! Thanks

File: contract_split/models/contract.py (hunk 1 above)
Comment (damdam-s) [reply contract_split/models/contract.py:53]: yes and if the line is completly moved from one contract to another, there is no origin contract line

**Pull request merged**